src/input/source/led/multicolor.rs

```diff
@@ -0,0 +1,272 @@
+        // Scale brightness to max_brightness
+        let brightness = brightness * 255 / self.max_brightness;
+        Ok(fs::write(
+            self.brightness_path.as_path(),
+            brightness.to_string().as_str(),
+        )
+        .map_err(|err| Box::new(LedMcError::Write(err)))?)
+    }
+
+    /// Reads the brightness path and returns a u8 with the current brightness setting
+    fn read_brightness(&self) -> Result<u8, Box<dyn Error + Send + Sync>> {
+        read_brightness(self.brightness_path.as_path())
+    }
+}
+
+impl Debug for LedMultiColor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("MultiColorChassis").finish()
+    }
+}
+
+impl SourceInputDevice for LedMultiColor {
+    fn poll(&mut self) -> Result<Vec<crate::input::event::native::NativeEvent>, InputError> {
+        Ok(Vec::new())
+    }
+
+    fn get_capabilities(&self) -> Result<Vec<Capability>, InputError> {
+        Ok(Vec::new())
+    }
+}
+
+impl SourceOutputDevice for LedMultiColor {
+    fn write_event(&mut self, event: OutputEvent) -> Result<(), OutputError> {
+        log::trace!("Received output event: {event:?}");
+        match event {
+            OutputEvent::DualSense(report) => {
+                if !report.allow_led_color {
+                    return Ok(());
+                }
+                self.write_color(report.led_red, report.led_green, report.led_blue)?;
+                self.write_brightness(255)?; // Assume max brightness.  TODO: Don't
+            }
+            _ => {
+                return Ok(());
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Reads the multi_intensity path and returns an LedFixedColor with the current color setting
+fn read_color(
+    path: &Path,
+    index: &[IndexType],
+) -> Result<LedFixedColor, Box<dyn Error + Send + Sync>> {
+    //TODO: Actually read this from supplied path.
+    let mut current_color: LedFixedColor = LedFixedColor::new(0, 0, 0);
+
+    let color_contents = read_to_string(path).map_err(|err| Box::new(LedMcError::Read(err)))?;
+    let color_vals = color_contents
+        .split_whitespace()
+        .map(|val| val.to_string().parse().unwrap_or_default())
+        .collect::<Vec<u32>>();
+
+    for (pos, e) in index.iter().enumerate() {
+        match e {
+            IndexType::RGB => {
+                let rgb: [u8; 4] = color_vals[pos].to_be_bytes();
+                log::debug!("Byte array: {rgb:?}");
+                // 0th index is null
+                current_color.r = rgb[1];
+                current_color.g = rgb[2];
+                current_color.b = rgb[3];
+                break; //TODO: Support more than one color for this type.
+            }
+            IndexType::Red => current_color.r = color_vals[pos] as u8,
+            IndexType::Green => current_color.g = color_vals[pos] as u8,
+            IndexType::Blue => current_color.b = color_vals[pos] as u8,
+        }
+    }
+
+    Ok(current_color)
```

Comment:
Instead of returning a hard-coded RGB value, we should instead return a vector of color values (i.e. `Vec<u8>`)